sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -848,6 +852,45 @@ def execute_query_raw(self, query):
             cursor.execute(query)
             return cursor.fetchall()
 
+    # https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-db-index-usage-stats-transact-sql?view=sql-server-ver15
+    def collect_index_usage_metrics(self):
+        if not is_affirmative(self.instance.get('include_index_usage_metrics', True)):
+            return False
+
+        now = time.time()
+        if now - self.index_usage_last_check_ts > INDEX_USAGE_STATS_INTERVAL:
+            self.log.warning('Collecting index usage statistics')
+            db_names = [d.name for d in self.databases] or [
+                self.instance.get('database', self.connection.DEFAULT_DATABASE)
+            ]
+            with self.connection.get_managed_cursor() as cursor:
+                try:
+                    for database in db_names:
+
+                        def query_executor(query, db=database):
+                            ctx = construct_use_statement(db)
+                            self.log.warning("changing cursor context via use statement: %s", ctx)
+                            cursor.execute(ctx)
```

Comment:
Is this connection reused? Do we need to have a `finally` block where we switch back to the default database?